File: android/wear/src/main/java/com/houari/yoroi/wear/presentation/SettingsPage.kt
package com.houari.yoroi.wear.presentation

import androidx.compose.foundation.background
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.wear.compose.foundation.lazy.ScalingLazyColumn
import androidx.wear.compose.material3.Text
import com.houari.yoroi.wear.data.YoroiDataRepository
import com.houari.yoroi.wear.theme.*

/**
 * Page 5: Reglages - Connexion, Timer Presets (horizontal scroll), Objectifs, Preferences
 */
@Composable
fun SettingsPage(repo: YoroiDataRepository) {
    val accent = remember(repo.themeAccentHex) { parseHexColor(repo.themeAccentHex) }
    val colors = rememberSyncedWatchColors(
        bgHex = repo.themeBgHex, cardBgHex = repo.themeCardBgHex,
        textPrimaryHex = repo.themeTextPrimaryHex, textSecondaryHex = repo.themeTextSecondaryHex,
        dividerHex = repo.themeDividerHex, textOnAccentHex = repo.themeTextOnAccentHex,
        isDarkMode = repo.isDarkMode
    )

    ScalingLazyColumn(
        modifier = Modifier.fillMaxSize(),
        contentPadding = PaddingValues(horizontal = 8.dp, vertical = 24.dp),
        verticalArrangement = Arrangement.spacedBy(6.dp)
    ) {
        // Header
        item {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                modifier = Modifier.padding(horizontal = 4.dp)
            ) {
                androidx.compose.material3.Icon(
                    Icons.Filled.Settings,
                    contentDescription = null,
                    tint = accent,
                    modifier = Modifier.size(12.dp)
                )
                Spacer(Modifier.width(6.dp))
                Text("Reglages", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = colors.textPrimary)
            }
        }

        // ── APPARENCE ──
        item { SectionLabel("APPARENCE", Icons.Filled.Palette, accent, colors) }
        item {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(10.dp))
                    .background(colors.cardBg)
                    .padding(8.dp),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    androidx.compose.material3.Icon(
                        if (repo.isDarkMode) Icons.Filled.DarkMode else Icons.Filled.LightMode,
                        contentDescription = null,
                        tint = if (repo.isDarkMode) Purple else Color(0xFFF97316),
                        modifier = Modifier.size(12.dp)
                    )
                    Spacer(Modifier.width(6.dp))
                    Text("Theme", fontSize = 10.sp, color = colors.textSecondary)
                    Spacer(Modifier.weight(1f))
                    Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                        ModeButton("Clair", Icons.Filled.LightMode, !repo.isDarkMode, Color(0xFFF97316), accent, colors) {
                            repo.changeThemeMode("light")
                        }
                        ModeButton("Sombre", Icons.Filled.DarkMode, repo.isDarkMode, Purple, accent, colors) {
                            repo.changeThemeMode("dark")
                        }
                    }
                }
            }
        }

        // ── CONNEXION ──
        item { SectionLabel("CONNEXION", Icons.Filled.Wifi, Green, colors) }
        item {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(10.dp))
                    .background(colors.cardBg)
                    .padding(8.dp),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(
                        modifier = Modifier
                            .size(8.dp)
                            .clip(CircleShape)
                            .background(if (repo.isConnected) Green else Red)
                    )
                    Spacer(Modifier.width(6.dp))
                    Text(
                        if (repo.isConnected) "Connecte" else "Deconnecte",
                        fontSize = 11.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = if (repo.isConnected) Green else Red
                    )
                }

                // Sync button
                androidx.compose.material3.Button(
                    onClick = { repo.requestSync() },
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Row(
                        horizontalArrangement = Arrangement.Center,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        androidx.compose.material3.Icon(
                            Icons.Filled.Sync,
                            contentDescription = null,
                            modifier = Modifier.size(12.dp)
                        )
                        Spacer(Modifier.width(4.dp))
                        Text("Synchroniser", fontSize = 10.sp, fontWeight = FontWeight.Bold)
                    }
                }
            }
        }

        // ── MINUTEUR (horizontal scroll presets) ──
        item { SectionLabel("MINUTEUR", Icons.Filled.Timer, accent, colors) }
        item {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(10.dp))
                    .background(colors.cardBg)
                    .padding(8.dp),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                Text("Duree par defaut", fontSize = 9.sp, color = colors.textSecondary)

                // Horizontal scrollable presets
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .horizontalScroll(rememberScrollState()),
                    horizontalArrangement = Arrangement.spacedBy(6.dp)
                ) {
                    listOf(
                        30 to "30s", 45 to "45s", 60 to "1:00", 90 to "1:30",
                        120 to "2:00", 150 to "2:30", 180 to "3:00", 240 to "4:00", 300 to "5:00"
                    ).forEach { (seconds, label) ->
                        val isSelected = repo.timerTotalSeconds == seconds && !repo.timerIsRunning
                        Box(
                            contentAlignment = Alignment.Center,
                            modifier = Modifier
                                .size(40.dp)
                                .clip(CircleShape)
                                .background(
                                    if (isSelected) accent
                                    else accent.copy(alpha = 0.1f)
                                )
                        ) {
                            androidx.compose.material3.TextButton(
                                onClick = { repo.setTimer(seconds); repo.changeTimerPreset(seconds) },
                                contentPadding = PaddingValues(0.dp),
                                modifier = Modifier.size(40.dp)
                            ) {
                                Text(
                                    label,
                                    fontSize = 9.sp,
                                    fontWeight = if (isSelected) FontWeight.Bold else FontWeight.Normal,
                                    color = if (isSelected) colors.textOnAccent else accent
                                )
                            }
                        }
                    }
                }
            }
        }

        // ── OBJECTIFS ──
        item { SectionLabel("OBJECTIFS", Icons.Filled.Flag, Green, colors) }
        item {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(10.dp))
                    .background(colors.cardBg)
                    .padding(8.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                // Steps goal
                GoalRow(
                    icon = Icons.Filled.DirectionsWalk,
                    color = Green,
                    label = "Pas / jour",
                    value = "${repo.stepsGoal}",
                    colors = colors,
                    onMinus = { repo.adjustStepsGoal(-1000) },
                    onPlus = { repo.adjustStepsGoal(1000) }
                )

                // Hydration goal
                GoalRow(
                    icon = Icons.Filled.WaterDrop,
                    color = Cyan,
                    label = "Eau / jour",
                    value = "${repo.hydrationGoal}ml",
                    colors = colors,
                    onMinus = { repo.adjustHydrationGoal(-250) },
                    onPlus = { repo.adjustHydrationGoal(250) }
                )
            }
        }

        // ── A PROPOS ──
        item { SectionLabel("A PROPOS", Icons.Filled.Info, colors.textSecondary, colors) }
        item {
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(10.dp))
                    .background(colors.cardBg)
                    .padding(8.dp),
                verticalArrangement = Arrangement.spacedBy(2.dp)
            ) {
                InfoRow("App", "Yoroi Watch", colors)
                InfoRow("Version", "2.0.0", colors)
                InfoRow("Plateforme", "WearOS", colors)
            }
        }
    }
}

@Composable
private fun SectionLabel(text: String, icon: ImageVector, color: Color, colors: YoroiWatchColors) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        modifier = Modifier.padding(horizontal = 4.dp)
    ) {
        androidx.compose.material3.Icon(
            icon,
            contentDescription = null,
            tint = color,
            modifier = Modifier.size(8.dp)
        )
        Spacer(Modifier.width(4.dp))
        Text(
            text,
            fontSize = 7.sp,
            fontWeight = FontWeight.ExtraBold,
            color = colors.textSecondary,
            letterSpacing = 1.sp
        )
    }
}

@Composable
private fun GoalRow(
    icon: ImageVector,
    color: Color,
    label: String,
    value: String,
    colors: YoroiWatchColors,
    onMinus: () -> Unit,
    onPlus: () -> Unit
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        verticalAlignment = Alignment.CenterVertically
    ) {
        androidx.compose.material3.Icon(
            icon,
            contentDescription = null,
            tint = color,
            modifier = Modifier.size(12.dp)
        )
        Spacer(Modifier.width(6.dp))
        Text(label, fontSize = 10.sp, color = colors.textSecondary)
        Spacer(Modifier.weight(1f))

        androidx.compose.material3.IconButton(
            onClick = onMinus,
            modifier = Modifier.size(24.dp)
        ) {
            androidx.compose.material3.Icon(
                Icons.Filled.RemoveCircle,
                contentDescription = null,
                tint = Red,
                modifier = Modifier.size(16.dp)
            )
        }

        Text(
            value,
            fontSize = 11.sp,
            fontWeight = FontWeight.Bold,
            color = colors.textPrimary,
            modifier = Modifier.widthIn(min = 48.dp)
        )

        androidx.compose.material3.IconButton(
            onClick = onPlus,
            modifier = Modifier.size(24.dp)
        ) {
            androidx.compose.material3.Icon(
                Icons.Filled.AddCircle,
                contentDescription = null,
                tint = Green,
                modifier = Modifier.size(16.dp)
            )
        }
    }
}

@Composable
private fun InfoRow(label: String, value: String, colors: YoroiWatchColors) {
    Row(modifier = Modifier.fillMaxWidth()) {
        Text(label, fontSize = 9.sp, color = colors.textSecondary)
        Spacer(Modifier.weight(1f))
        Text(value, fontSize = 9.sp, fontWeight = FontWeight.Bold, color = colors.textPrimary)
    }
}

@Composable
private fun ModeButton(
    label: String,
    icon: ImageVector,
    isSelected: Boolean,
    iconColor: Color,
    accent: Color,
    colors: YoroiWatchColors,
    onClick: () -> Unit
) {
    androidx.compose.material3.TextButton(
        onClick = onClick,
        contentPadding = PaddingValues(horizontal = 6.dp, vertical = 3.dp),
        modifier = Modifier
            .clip(RoundedCornerShape(6.dp))
            .background(if (isSelected) accent else colors.cardBg)
    ) {
        Row(
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(3.dp)
        ) {
            androidx.compose.material3.Icon(
                icon,
                contentDescription = null,
                tint = if (isSelected) iconColor else colors.textSecondary,
                modifier = Modifier.size(10.dp)
            )
            Text(
                label,
                fontSize = 8.sp,
                fontWeight = FontWeight.Bold,
                color = if (isSelected) colors.textOnAccent else colors.textPrimary
            )
        }
    }
}

File: ios/YoroiWatch/Views/SettingsPage.swift
import SwiftUI
import WatchKit

// ============================================================
// PAGE 5: REGLAGES
// ============================================================

struct SettingsPage: View {
  @EnvironmentObject var session: WatchSessionManager
  @AppStorage("defaultTimerSeconds") private var defaultTimerSeconds: Int = 90
  @AppStorage("hapticEnabled") private var hapticEnabled: Bool = true
  @AppStorage("unitSystem") private var unitSystem: String = "kg" // kg or lbs

  @State private var showTimerPicker = false

  // Timer presets for picker
  private let timerOptions: [(Int, String)] = [
    (30, "30s"), (45, "45s"), (60, "1:00"), (90, "1:30"),
    (120, "2:00"), (150, "2:30"), (180, "3:00"), (240, "4:00"), (300, "5:00")
  ]

  var body: some View {
    ScrollView {
      VStack(spacing: 10) {

        // ── HEADER ──
        HStack(spacing: 6) {
          Image(systemName: "gearshape.fill")
            .font(.system(size: 12))
            .foregroundColor(goldColor)
          Text("Reglages")
            .font(.system(size: 14, weight: .bold))
            .foregroundColor(session.textPrimary)
          Spacer()
        }
        .padding(.horizontal, 4)

        // ── APPARENCE ──
        VStack(spacing: 4) {
          sectionLabel("APPARENCE", icon: "paintbrush.fill", color: goldColor)

          HStack {
            Image(systemName: session.isDarkMode ? "moon.fill" : "sun.max.fill")
              .font(.system(size: 10))
              .foregroundColor(session.isDarkMode ? .purple : .orange)
            Text("Theme")
              .font(.system(size: 10))
              .foregroundColor(session.textSecondary)
            Spacer()
            HStack(spacing: 4) {
              modeButton("Clair", icon: "sun.max.fill", isSelected: !session.isDarkMode, iconColor: .orange)
              modeButton("Sombre", icon: "moon.fill", isSelected: session.isDarkMode, iconColor: .purple)
            }
          }
          .padding(.horizontal, 6)
          .padding(.vertical, 4)
        }

        // ── CONNEXION ──
        VStack(spacing: 4) {
          sectionLabel("CONNEXION", icon: "antenna.radiowaves.left.and.right", color: .green)

          HStack {
            Circle()
              .fill(session.isConnected ? Color.green : Color.red)
              .frame(width: 8, height: 8)
            Text(session.isConnected ? "Connecte" : "Deconnecte")
              .font(.system(size: 11, weight: .semibold))
              .foregroundColor(session.isConnected ? .green : .red)
            Spacer()
            if let lastSync = session.lastSyncDate {
              Text(timeAgo(lastSync))
                .font(.system(size: 8))
                .foregroundColor(session.textSecondary)
            }
          }
          .padding(.horizontal, 6)

          Button(action: {
            session.requestSync()
            if hapticEnabled { WKInterfaceDevice.current().play(.click) }
          }) {
            HStack(spacing: 4) {
              Image(systemName: "arrow.triangle.2.circlepath")
                .font(.system(size: 10))
              Text("Synchroniser")
                .font(.system(size: 10, weight: .bold))
            }
            .foregroundColor(session.textOnAccent)
            .frame(maxWidth: .infinity)
            .padding(.vertical, 7)
            .background(goldColor)
            .cornerRadius(8)
          }
          .buttonStyle(.plain)
        }

        // ── MINUTEUR ──
        VStack(spacing: 4) {
          sectionLabel("MINUTEUR", icon: "timer", color: goldColor)

          Button(action: { showTimerPicker = true }) {
            HStack {
              Text("Duree par defaut")
                .font(.system(size: 10))
                .foregroundColor(session.textSecondary)
              Spacer()
              Text(formatTime(defaultTimerSeconds))
                .font(.system(size: 12, weight: .bold))
                .foregroundColor(goldColor)
              Image(systemName: "chevron.right")
                .font(.system(size: 8))
                .foregroundColor(session.textSecondary.opacity(0.5))
            }
            .padding(.horizontal, 6)
            .padding(.vertical, 6)
            .background(session.cardBg)
            .cornerRadius(8)
          }
          .buttonStyle(.plain)
        }

        // ── OBJECTIFS ──
        VStack(spacing: 4) {
          sectionLabel("OBJECTIFS", icon: "target", color: .green)

          // Steps goal
          HStack {
            Image(systemName: "figure.walk")
              .font(.system(size: 10))
              .foregroundColor(.green)
            Text("Pas / jour")
              .font(.system(size: 10))
              .foregroundColor(session.textSecondary)
            Spacer()
            Button(action: {
              session.updateStepsGoal(max(1000, session.stepsGoal - 1000))
              if hapticEnabled { WKInterfaceDevice.current().play(.click) }
            }) {
              Image(systemName: "minus.circle.fill")
                .font(.system(size: 14))
                .foregroundColor(.red)
            }.buttonStyle(.plain)
            Text("\(session.stepsGoal)")
              .font(.system(size: 12, weight: .bold))
              .foregroundColor(session.textPrimary)
              .frame(width: 44)
            Button(action: {
              session.updateStepsGoal(min(30000, session.stepsGoal + 1000))
              if hapticEnabled { WKInterfaceDevice.current().play(.click) }
            }) {
              Image(systemName: "plus.circle.fill")
                .font(.system(size: 14))
                .foregroundColor(.green)
            }.buttonStyle(.plain)
          }
          .padding(.horizontal, 6)
          .padding(.vertical, 4)

          // Hydration goal
          HStack {
            Image(systemName: "drop.fill")
              .font(.system(size: 10))
              .foregroundColor(.cyan)
            Text("Eau / jour")
              .font(.system(size: 10))
              .foregroundColor(session.textSecondary)
            Spacer()
            Button(action: {
              session.updateHydrationGoal(max(500, session.hydrationGoal - 250))
              if hapticEnabled { WKInterfaceDevice.current().play(.click) }
            }) {
              Image(systemName: "minus.circle.fill")
                .font(.system(size: 14))
                .foregroundColor(.red)
            }.buttonStyle(.plain)
            Text("\(session.hydrationGoal)ml")
              .font(.system(size: 11, weight: .bold))
              .foregroundColor(session.textPrimary)
              .frame(width: 52)
            Button(action: {
              session.updateHydrationGoal(min(6000, session.hydrationGoal + 250))
              if hapticEnabled { WKInterfaceDevice.current().play(.click) }
            }) {
              Image(systemName: "plus.circle.fill")
                .font(.system(size: 14))
                .foregroundColor(.green)
            }.buttonStyle(.plain)
          }
          .padding(.horizontal, 6)
          .padding(.vertical, 4)
        }

        // ── PREFERENCES ──
        VStack(spacing: 4) {
          sectionLabel("PREFERENCES", icon: "slider.horizontal.3", color: .purple)

          // Unit system
          HStack {
            Text("Unites")
              .font(.system(size: 10))
              .foregroundColor(session.textSecondary)
            Spacer()
            HStack(spacing: 4) {
              unitButton("kg", isSelected: unitSystem == "kg")
              unitButton("lbs", isSelected: unitSystem == "lbs")
            }
          }
          .padding(.horizontal, 6)
          .padding(.vertical, 4)

          // Haptic
          HStack {
            Text("Vibrations")
              .font(.system(size: 10))
              .foregroundColor(session.textSecondary)
            Spacer()
            Button(action: {
              hapticEnabled.toggle()
              if hapticEnabled { WKInterfaceDevice.current().play(.click) }
            }) {
              Image(systemName: hapticEnabled ? "iphone.radiowaves.left.and.right" : "iphone.slash")
                .font(.system(size: 12))
                .foregroundColor(hapticEnabled ? .green : session.textSecondary)
                .frame(width: 28, height: 28)
                .background(hapticEnabled ? Color.green.opacity(0.15) : session.cardBg)
                .cornerRadius(7)
            }
            .buttonStyle(.plain)
          }
          .padding(.horizontal, 6)
          .padding(.vertical, 4)
        }

        // ── A PROPOS ──
        VStack(spacing: 4) {
          sectionLabel("A PROPOS", icon: "info.circle.fill", color: .gray)

          VStack(spacing: 2) {
            infoRow("App", value: "Yoroi Watch")
            infoRow("Version", value: "2.0.0")
            infoRow("Exercices", value: "\(ExerciseTemplate.library.count)")
          }
          .padding(.horizontal, 6)
          .padding(.vertical, 4)
          .background(session.cardBg)
          .cornerRadius(8)
        }

        // Spacer at bottom
        Spacer().frame(height: 20)
      }
      .padding(.horizontal, 2)
    }
    .refreshable { session.requestSync() }
    .sheet(isPresented: $showTimerPicker) {
      timerPickerSheet
    }
  }

  // MARK: - Timer Picker Sheet
  private var timerPickerSheet: some View {
    ScrollView {
      VStack(spacing: 6) {
        Text("DUREE PAR DEFAUT")
          .font(.system(size: 8, weight: .heavy))
          .foregroundColor(goldColor)
          .tracking(1)

        ForEach(timerOptions, id: \.0) { (seconds, label) in
          Button(action: {
            defaultTimerSeconds = seconds
            session.setTimer(seconds: seconds)
            session.changeTimerPreset(seconds)
            showTimerPicker = false
            if hapticEnabled { WKInterfaceDevice.current().play(.click) }
          }) {
            HStack {
              Text(label)
                .font(.system(size: 13, weight: .bold))
                .foregroundColor(seconds == defaultTimerSeconds ? session.textOnAccent : session.textPrimary)
              Spacer()
              if seconds == defaultTimerSeconds {
                Image(systemName: "checkmark")
                  .font(.system(size: 10, weight: .bold))
                  .foregroundColor(session.textOnAccent)
              }
            }
            .padding(.horizontal, 10)
            .padding(.vertical, 8)
            .background(seconds == defaultTimerSeconds ? goldColor : session.cardBg)
            .cornerRadius(8)
          }
          .buttonStyle(.plain)
        }
      }
      .padding(.horizontal, 4)
    }
  }

  // MARK: - Helpers

  private func sectionLabel(_ text: String, icon: String, color: Color) -> some View {
    HStack(spacing: 4) {
      Image(systemName: icon)
        .font(.system(size: 8))
        .foregroundColor(color)
      Text(text)
        .font(.system(size: 7, weight: .heavy))
        .foregroundColor(session.textSecondary)
        .tracking(1)
      Spacer()
    }
    .padding(.horizontal, 4)
  }

  private func modeButton(_ label: String, icon: String, isSelected: Bool, iconColor: Color) -> some View {
    Button(action: {
      let mode = label == "Clair" ? "light" : "dark"
      session.changeThemeMode(mode)
      if hapticEnabled { WKInterfaceDevice.current().play(.click) }
    }) {
      HStack(spacing: 3) {
        Image(systemName: icon)
          .font(.system(size: 8))
          .foregroundColor(isSelected ? iconColor : session.textSecondary)
        Text(label)
          .font(.system(size: 8, weight: .bold))
          .foregroundColor(isSelected ? session.textOnAccent : session.textPrimary)
      }
      .padding(.horizontal, 8)
      .padding(.vertical, 5)
      .background(isSelected ? goldColor : session.cardBg)
      .cornerRadius(6)
    }
    .buttonStyle(.plain)
  }

  private func unitButton(_ unit: String, isSelected: Bool) -> some View {
    Button(action: {
      unitSystem = unit
      session.changeUnitSystem(unit)
      if hapticEnabled { WKInterfaceDevice.current().play(.click) }
    }) {
      Text(unit.uppercased())
        .font(.system(size: 9, weight: .bold))
        .foregroundColor(isSelected ? session.textOnAccent : session.textPrimary)
        .padding(.horizontal, 10)
        .padding(.vertical, 5)
        .background(isSelected ? goldColor : session.cardBg)
        .cornerRadius(6)
    }
    .buttonStyle(.plain)
  }

  private func infoRow(_ label: String, value: String) -> some View {
    HStack {
      Text(label)
        .font(.system(size: 9))
        .foregroundColor(session.textSecondary)
      Spacer()
      Text(value)
        .font(.system(size: 9, weight: .bold))
        .foregroundColor(session.textPrimary)
    }
    .padding(.vertical, 1)
  }

  private func formatTime(_ seconds: Int) -> String {
    let m = seconds / 60
    let s = seconds % 60
    if s == 0 { return "\(m) min" }
    return "\(m):\(String(format: "%02d", s))"
  }

  private func timeAgo(_ date: Date) -> String {
    let diff = Date().timeIntervalSince(date)
    if diff < 60 { return "a l'instant" }
    if diff < 3600 { return "il y a \(Int(diff / 60))min" }
    return "il y a \(Int(diff / 3600))h"
  }

  private var goldColor: Color {
    session.accentColor
  }
}
